python/lsst/pipe/base/_dataset_handle.py

```diff
@@ -127,14 +127,14 @@ def _getStorageClass(self) -> StorageClass:
 
         # Need to match python type.
         pytype = type(self.inMemoryDataset)
-        for storageClass in factory.values():
-            # It is possible for a single python type to refer to multiple
-            # storage classes such that this could be quite fragile.
-            if storageClass.is_type(pytype):
-                return storageClass
+
+        try:
+            return factory.findStorageClass(pytype)
+        except KeyError:
+            pass
```

Comment:
Would this be clearer if it just raised the ValueError here?

Reply:
I don't like raising inside an `except` block unless that's really necessary because of all the exception chaining that's not really relevant to the caller. A clean exception outside seemed better. The alternative is to say that this method actually raises `KeyError` and let the exception survive as is.